cpp/src/arrow/ipc/test-common.h

```diff
@@ -223,15 +223,17 @@ Status MakeRandomBinaryArray(int64_t length, bool include_nulls, MemoryPool* poo
     if (include_nulls && values_index == 0) {
       RETURN_NOT_OK(builder.AppendNull());
     } else {
-      const std::string& value = values[values_index];
+      const std::string value =
+          i < int64_t(values.size()) ? values[values_index] : std::to_string(i);
       RETURN_NOT_OK(builder.Append(reinterpret_cast<const RawType*>(value.data()),
                                    static_cast<int32_t>(value.size())));
     }
   }
   return builder.Finish(out);
 }
 
-Status MakeStringTypesRecordBatch(std::shared_ptr<RecordBatch>* out) {
+Status MakeStringTypesRecordBatchWithNulls(std::shared_ptr<RecordBatch>* out,
```

Comment:
Style note: if it's called `MakeStringTypesRecordBatchWithNulls`, it shouldn't take an argument indicating whether you want to include nulls. Consider renaming or using a private helper function.

Reply:
Or simply add a `bool with_nulls = true` optional argument to `MakeStringTypesRecordBatch`.

Reply:
Adding the default argument didn't work, because MakeStringTypesRecordBatch is used as a function pointer in other tests. I'll rename is to MakeStringTypesRecordBatchWithoutNulls abd this one to MakeStringTypesRecordBatch.